tests/SAML2/XML/alg/SigningMethodTest.php

```diff
@@ -34,25 +33,10 @@ public function testMarshalling(): void
         );
         $this->assertCount(1, $signingMethodElements);
         $signingMethodElement = $signingMethodElements[0];
+
         $this->assertEquals('http://exampleAlgorithm', $signingMethodElement->getAttribute('Algorithm'));
-        $this->assertFalse($signingMethodElement->hasAttribute('MinKeySize'));
+        $this->assertEquals('1024', $signingMethodElement->getAttribute('MinKeySize'));
```

Comment:
Should we be a bit more exhaustive with the tests? As in, testing all possible scenarios of missing (or not) attributes...

Reply:
We used to do all attributes set and optional attributes missing.. I reduced it to a mixed scenario, but I can easily revert that..